firewood/src/db.rs

```diff
@@ -272,6 +273,43 @@ pub struct DbRev<S> {
     merkle: Merkle<S>,
 }
 
+#[async_trait]
+impl<S: ShaleStore<Node> + Send + Sync> api::DbView for DbRev<S> {
+    async fn root_hash(&self) -> Result<api::HashKey, api::Error> {
+        self.merkle
+            .root_hash(self.header.kv_root)
+            .map(|h| *h)
+            .map_err(|e| api::Error::IO(std::io::Error::new(ErrorKind::Other, e)))
+    }
+
+    async fn val<K: api::KeyType>(&self, key: K) -> Result<Option<Vec<u8>>, api::Error> {
+        let obj_ref = self.merkle.get(key, self.header.kv_root);
+        match obj_ref {
+            Err(e) => Err(api::Error::IO(std::io::Error::new(ErrorKind::Other, e))),
+            Ok(obj) => match obj {
+                None => Ok(None),
+                Some(inner) => Ok(Some(inner.as_ref().to_owned())),
```

Comment:
nit: Naming is hard.., we discussed to no longer use `inner` as it didn't bring any clarification?

Reply:
I think we need to stop using the `inner` pattern. Naming something `inner` when we've simply added a wrapper type is fine IMO

Reply:
Yeah. This is one of those sweep tasks we can assign as a good first issue. Anyone want to create one?

Reply:
I don't think it's trivial to unravel all the ` SomeType { inner: SomeInnerType }` especially with all the misused trait-bounds